### ISSUE TITLE
LightGBMTuner selects last metric and validation set for optimization

### ISSUE THREAD

**Issue by rabyj** [optuna/optuna#3994]
**Disclaimer**: I do not know if any of what I experienced is considered a bug for optuna.

**Summary**: The behaviour of [LightGBMTuner](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.lightgbm.LightGBMTuner.html#optuna.integration.lightgbm.LightGBMTuner) is not specified for multimetric and multiple validation sets.

**Details**: If the `params` dictionary contains a "metric" parameter with a list of metrics, the last one will be selected during the optimization process as the objective score to optimize, not the default metric linked to the "objective" parameter. I think this behaviour should be documented, as the `metric` entry is used for logging and evaluation purposes and such in LightGBM, and it is not linked to an optimization process.

In the following example, `auc_mu` would be used for optimization, even though the LightGBM objective is `multi_logloss`

```python
params = {
    "objective": "multiclass",
    "num_class": 4,
    "metric": ["multi_error", "multi_logloss", "auc_mu"],
    "boosting_type": "dart",
    "seed": 42,
    "force_col_wise": True,
}
```

Furthermore, the metric chosen for the optimization process is also affected by the order of the `valid_sets` given. If there are several, it will choose the last set in the list for the objective score. I think this behaviour should be specified in `valid_sets (Optional[VALID_SET_TYPE]) –`. This last behaviour is also described more clearly in #1984 (which was never addressed)

I do not know if any of the two mentioned situations is expected behaviour, but if it is, it should be specified clearly somewhere, as it is unintuitive. For example, I thought my tuner would minimize the `multi_logloss` since I had specified `"objective": "multiclass"`, but it tried to minimize `auc_mu` because it was the last metric in the list (and on the training set instead of validation!!!). It was really confusing before I understood this strange behaviour.

**Comment by rabyj** [comment]
@jeffzi any news on this old issue? Related #1984 and PR #1947 went stale without improvements to the doc.

**Comment by rabyj** [comment]
I think this is still relevant even though it has no attention
